0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,353 @@
+Accept: application/json
+Content-Type: application/json
+```
+
+> **[`Quantity`](#quantity-type)** to settle
+>
+> - If the amount is `0`, the settlement engine MAY retry failed or queued settlements.
+
+##### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+```
+
+> **[`Quantity`](#quantity-type)** enqueued to settle
+>
+> - This response does not guarantee that a settlement was executed.
+> - If the settlement engine uses a unit less precise than the accounting system's unit, in order to prevent the two systems getting out-of-sync, it must indicate the amount it queued so the accounting system can correctly track the amount leftover.
+
+#### Handle incoming request
+
+Respond to an incoming message from the given peer's settlement engine. The transport MUST be the only entity invoking this instruction.
+
+##### Request
+
+```http
+POST /accounts/:id/handleMessage HTTP/1.1
+Accept: application/octet-stream
+Content-Type: application/octet-stream
+```
+
+> _&lt;raw bytes of message&gt;_
+
+##### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/octet-stream
+```
+
+> _&lt;raw bytes of response&gt;_
+
+#### Complete specification
+
+<details>
+  <summary>The full settlement engine HTTP API, including all response codes, error codes, and additional endpoints for manual management are available in this Swagger definition.</summary>
+
+```
+TODO Embed the definition here
+```
+
+</details>
+
+Settlement engine implementations are RECOMMENDED to implement the standardized endpoints for manual management noted above.
+
+#### Additional RPCs
+
+Settlement engine implementations MAY expose additional, non-standard endpoints for manual operations or configuration.
+
+For example, settlement engines that use payment channels may expose functionality to manually open, fund, and close channels.
+
+### Accounting System HTTP API
+
+#### Credit incoming settlement
+
+Notify the accounting system of receipt of an incoming settlement. Note that the settlement engine MAY accrue incoming settlement acknowledgements without immediately informing the accounting system.
+
+The settlement engine MUST be the only entity invoking this message and MUST credit this amount to the accounts receivable, or some asset account tracking the balance owed by the peer, to ensure accurate accounting.
+
+##### Request
+
+```http
+POST /accounts/:id/receipt HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+```
+
+> **[`Quantity`](#quantity-type)** that should be credited to the account as an incoming settlement
+
+##### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+```
+
+> **[`Quantity`](#quantity-type)** credited to the account
+>
+> - If the accounting system uses a unit less precise than the settlement engine's unit in order to prevent the two systems getting out-of-sync, it must indicate the amount it acknowledged receipt of so the settlement engine can track the amount leftover.
```

Comment:
As above, this response should always be sent

Reply:
Updated here: https://github.com/interledger/rfcs/compare/4e742ae54dcd738b67a4a541f9070837a3c16e98...ko-settlement-engine#diff-990a915317f4e35578edfa70631bd07dL287